experiments/snippets/skuidCustom/skuidCustom.js

```diff
@@ -0,0 +1,1741 @@
+		const localLimit = fparams.limit || model.recordsLimit || 200;
+		const exportWhenDone = fparams.exportWhenDone || false;
+		const exportOptions = fparams.exportOptions || {};
+		const progressCallback = fparams.progressCallback || undefined;
+
+		// Set Initial Progress
+		if (progressCallback !== undefined) {
+			progressCallback.call(this, {
+				count: 0,
+				limit: localLimit,
+				nextStart: 1,
+				nextEnd: localLimit,
+				model: model
+			});
+		}
+		model.abandonAllRows();
+
+		// Run through our recursive function to load all rows
+		modelLoadSOQL(model, {
+			limit: limit,
+			progressCallback: progressCallback,
+			promiseResolve: deferred.resolve,
+			promiseReject: deferred.reject,
+			exportWhenDone: exportWhenDone,
+			exportOptions: exportOptions
+		});
+	}
+
+	function modelLoadSOQL (model, fparams) {
+		fparams = fparams || {};
+		const limit = fparams.limit || model.recordsLimit || 200;
+		const last = fparams.last || undefined;
+		const count = fparams.count || 0;
+		let tempRows = fparams.tempRows || [];
+		const progressCallback = fparams.progressCallback || undefined;
+		const promiseResolve = fparams.promiseResolve || undefined;
+		const promiseReject = fparams.promiseReject || undefined;
+		const exportWhenDone = fparams.exportWhenDone || false;
+		const exportOptions = fparams.exportOptions || {};
+		const modelArrayPosition = fparams.modelArrayPosition || 0;
+		const modelArray = fparams.modelArray || undefined;
+		let limitSpecified = fparams.limitSpecified;
+
+		if (limitSpecified === undefined) {
+			if (fparams.limit !== undefined) {
+				limitSpecified = true;
+			} else {
+				limitSpecified = false;
+			}
+		}
+
+		const displayTypesToEnclose = ['STRING', 'ID', 'TEXT', 'TEXTAREA', 'PICKLIST', 'MULTIPICKLIST',
+			'PHONE', 'REFERENCE', 'URL', 'EMAIL', 'ADDRESS', 'ENCRYPTEDSTRING'];
+		// const displayTypesDontEnclose = ['DOUBLE', 'BOOLEAN', 'CURRENCY', 'DATE',
+		//	'DATETIME', 'INTEGER', 'PERCENT'];
+
+		let orderby = '';
+		let having = '';
+		let groupby = '';
+		const groupbyItemsArr = [];
+
+		let fieldsStr = '';
+
+		let noIdField = true;
+
+		// construct fields from model fields
+		if (model.fields !== undefined) {
+			for (let i = 0; i < model.fields.length; i++) {
+				const f = model.fields[i];
+
+				// skip UI only fields and non-querying fields
+				if (f.uiOnly === true || f.query === false) {
+					continue;
+				}
+
+				let fieldBefore = '';
+				let fieldAfter = '';
+				const field = f.id;
+
+				if (field === 'Id') {
+					noIdField = false;
+				}
+				
+				let fieldName = escapeSOQL(f.name) || '';
+				if (fieldName !== '') {
+					fieldName = ` ${fieldName}`;	
+				}
+				// remove name for non-aggregate models
+				if (model.isAggregate === false) {
+					fieldName = '';
+				}
+
+				if (fieldsStr !== '') {
+					fieldsStr += ',';
+				}
+
+				// If this is a child relationship field
+				if (f.type !== undefined && f.type === 'childRelationship') {
+
+				}

```

Comment:
Now that the linter is running on this file, it is complaining about this empty code block... can we just remove this?

Reply:
Ah -- the modelLoader currently breaks if there are child relationships, I have not yet implemented the child relationship code. That code there is a placeholder for a future addition. I've commented it out for now. Is that ok?